src/client/java/teammates/client/scripts/DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat.java

```diff
@@ -0,0 +1,205 @@
+        println("Running data migration for " + commentsToFix.size() + " documents in old date format...");
+        println("Preview: " + isPreview);
+
+        List<Document> documentsToUpdate = new ArrayList<>();
+
+        LoopHelper loopHelper = new LoopHelper(batchSize, "documents processed.");
+
+        for (FeedbackResponseCommentAttributes commentToFix : commentsToFix) {
+            loopHelper.recordLoop();
+            queueDocumentUpdate(commentToFix, documentsToUpdate);
+
+            if (documentsToUpdate.size() == batchSize) {
+                updateAndClearDocuments(documentsToUpdate);
+            }
+        }
+        updateAndClearDocuments(documentsToUpdate);
+
+        println("\nComplete! If there are any failures shown above, please rerun this script.");
+    }
+
+    private void queueDocumentUpdate(FeedbackResponseCommentAttributes comment, List<Document> documentsToUpdate) {
+        documentsToUpdate.add(new FeedbackResponseCommentSearchDocument(comment).build());
+    }
+
+    private void updateAndClearDocuments(List<Document> documentsToUpdate) {
+        if (documentsToUpdate.isEmpty()) {
+            return;
+        }
+
+        println("Batch updating " + documentsToUpdate.size() + " documents...");
+
+        if (!isPreview) {
+            try {
+                invokePutDocumentsWithRetry(Const.SearchIndex.FEEDBACK_RESPONSE_COMMENT, documentsToUpdate);
+                println("Batch update succeeded.");
+            } catch (PutException e) {
+                println("Batch update failed with non-transient errors.");
+            } catch (MaximumRetriesExceededException e) {
+                println("Batch update failed after maximum retries.");
+            }
+        }
+
+        documentsToUpdate.clear();
+    }
+
+    /**
+     * Reflects private static method {@link SearchManager#putDocumentsWithRetry}.
+     *
+     * @throws PutException when only non-transient errors are encountered.
+     * @throws MaximumRetriesExceededException with list of failed {@link Document}s as final data and
+     *         final {@link com.google.appengine.api.search.OperationResult}'s message as final message,
+     *         if operation fails after maximum retries.
+     */
+    private static void invokePutDocumentsWithRetry(String indexName, List<Document> documents)
+            throws PutException, MaximumRetriesExceededException {
+        try {
+            Method method = SearchManager.class.getDeclaredMethod("putDocumentsWithRetry", String.class, List.class);
+            method.setAccessible(true);
+            method.invoke(null, indexName, documents);
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        } catch (InvocationTargetException e) {
+            Throwable originalException = e.getCause();
+            if (originalException instanceof PutException) {
+                throw (PutException) originalException;
```

Comment:
Outside the scope of `invokePutDocumentsWithRetry`, which is only to invoke the private method?